test/csuite/wt6616_checkpoint_oldest_ts/main.c

```diff
@@ -0,0 +1,372 @@
+
+static const char *const uri = "table:wt6616-checkpoint-oldest-ts";
+
+static const char *const ckpt_file = "checkpoint_done";
+
+static bool log_table = false;
+
+#define ENV_CONFIG                                            \
+    "cache_size=50M,"                                         \
+    "create,"                                                 \
+    "eviction_updates_target=20,eviction_updates_trigger=90," \
+    "statistics=(fast),statistics_log=(wait=1,json=true),"    \
+    "timing_stress_for_test=[checkpoint_slow]"
+
+#define ENV_CONFIG_REC "log=(archive=false,recover=on)"
+
+static void handler(int) WT_GCC_FUNC_DECL_ATTRIBUTE((noreturn));
+static void usage(void) WT_GCC_FUNC_DECL_ATTRIBUTE((noreturn));
+static void
+usage(void)
+{
+    fprintf(stderr, "usage: %s [-h dir] [-t time]\n", progname);
+    exit(EXIT_FAILURE);
+}
+
+/*
+ * thread_ckpt_run --
+ *     Runner function for the checkpoint thread.
+ */
+static WT_THREAD_RET
+thread_ckpt_run(void *arg)
+{
+    FILE *fp;
+    WT_CONNECTION *conn;
+    WT_RAND_STATE rnd;
+    WT_SESSION *session;
+    uint64_t stable;
+    uint32_t sleep_time;
+    int i;
+    char ts_string[WT_TS_HEX_STRING_SIZE];
+    bool first_ckpt;
+
+    __wt_random_init(&rnd);
+
+    conn = (WT_CONNECTION *)arg;
+    /* Keep a separate file with the records we wrote for checking. */
+    (void)unlink(ckpt_file);
+    testutil_check(conn->open_session(conn, NULL, NULL, &session));
+    first_ckpt = true;
+    for (i = 0;; ++i) {
+        sleep_time = __wt_random(&rnd) % MAX_CKPT_INVL;
+        sleep(sleep_time);
+        /* Run checkpoint with timestamps. */
+        testutil_check(session->checkpoint(session, "use_timestamp=true"));
+        testutil_check(conn->query_timestamp(conn, ts_string, "get=last_checkpoint"));
+        testutil_assert(sscanf(ts_string, "%" SCNx64, &stable) == 1);
+        printf("Checkpoint %d complete at stable %" PRIu64 ".\n", i, stable);
+        fflush(stdout);
+        /*
+         * Create the checkpoint file so that the parent process knows at least one checkpoint has
+         * finished and can start its timer.
+         */
+        if (first_ckpt) {
+            testutil_checksys((fp = fopen(ckpt_file, "w")) == NULL);
+            first_ckpt = false;
+            testutil_checksys(fclose(fp) != 0);
+        }
+    }
+    /* NOTREACHED */
+}
+
+/*
+ * thread_run --
+ *     Runner function for the worker thread.
+ */
+static WT_THREAD_RET
+thread_run(void *arg)
+{
+    WT_CONNECTION *conn;
+    WT_CURSOR *cursor;
+    WT_ITEM data;
+    WT_SESSION *session;
+    uint64_t ts, oldest_ts;
+    char kname[64], tscfg[64];
+
+    conn = (WT_CONNECTION *)arg;
+
+    testutil_check(conn->open_session(conn, NULL, "isolation=snapshot", &session));
+    testutil_check(session->open_cursor(session, uri, NULL, NULL, &cursor));
+
+    /* Insert and then delete the keys until we're killed. */
+    printf("Worker thread started.\n");
+    for (ts = 1, oldest_ts = 0;; ++ts) {
```

Comment:
Alphabetize here too.

Also, do you want to increment `ts` by 2 since you're using ts + 1 for the delete each loop? It seems like it is fine either way, I'm just wondering if there is any benefit one way or the other.

Reply:
I think both works the same.